pkg/nfd-topology-updater/nfd-topology-updater-internal_test.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nfdtopologyupdater
+
+import (
+
+	"testing"
+
+  v1alpha1 "github.com/swatisehgal/topologyapi/pkg/apis/topology/v1alpha1"
+	. "github.com/smartystreets/goconvey/convey"
+	"github.com/stretchr/testify/mock"
+	"github.com/vektra/errors"
+	"sigs.k8s.io/node-feature-discovery/pkg/topologyupdater"
+)
+
+
+func TestNewTopologyUpdater(t *testing.T) {
+	Convey("When creating new TopologyUpdater instance", t, func() {
+    args := Args{}
+    tmPolicy := "fake-topology-manager-policy"
+
+		Convey("without any args and tmPolicy specified", func() {
+			u, err := NewTopologyUpdater(args, "")
+			Convey("no error should be returned", func() {
+				So(err, ShouldBeNil)
+			})
+			updater := u.(*nfdTopologyUpdater)
+			Convey("no sources should be enabled and the whitelist regexp should be empty", func() {
+				So(len(updater.tmPolicy), ShouldEqual, 0)
```

Comment:
we should never access/test the private methods

Reply:
Again, I thought so too that we should be testing only public function but noticed https://github.com/kubernetes-sigs/node-feature-discovery/blob/master/pkg/nfd-worker/nfd-worker-internal_test.go which got me thinking that NFD maintainers would be expecting this from us.